### PR TITLE
Added option to perform linear resampling

### PR DESCRIPTION
**Pull request by jnecus**
I found that I was using this in my work. It is a useful feature to have and easily implementable.

### CODE REVIEW

**Comment by jnecus** [comment]
I don't quite understand codecov and why this minor change has resulted in a failure. Can anybody help explain? :-)

**Comment by bthirion** [comment]
Looks good, thx; could you add a test to check that linear resampling behaves well.

**Comment by bthirion** [comment]
The "failure" is just that you added one untested line of code ;-)

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -375,10 +375,14 @@ def resample_img(img, target_affine=None, target_shape=None,
 
     if interpolation == 'continuous':
         interpolation_order = 3
+        
+    elif interpolation == 'linear':
+        interpolation_order = 1
+
```
Comment (GaelVaroquaux) [inline nilearn/image/resampling.py]: Style nitpick: could you remove the empty lines before the two elif

**Comment by GaelVaroquaux** [comment]
This is almost there. We need you to add a test of the new mode. Specifically, in the file nilearn/image/tests/test_resampling.py, you could add "linear" to the list of options tested at line 96 and line 102.

Thanks!

**Comment by jnecus** [comment]
@GaelVaroquaux Thank's for the direct advice. I'm new to this :-)

Should be all sorted now.

File: nilearn/image/tests/test_resampling.py [hunk 2]
```diff
@@ -92,14 +92,14 @@ def test_identity_resample():
     # Test with non native endian data
 
     # Test with big endian data ('>f8')
-    for interpolation in ['nearest', 'continuous']:
+    for interpolation in ['nearest', 'linear','continuous']:
```
Comment (GaelVaroquaux) [inline nilearn/image/tests/test_resampling.py]: Sorry to always come back with a new comment, but the Python style guide (known as 'PEP8') says that there should be a space here, after the coma: always a space after, never before.

File: nilearn/image/tests/test_resampling.py [hunk 3]
```diff
@@ -92,14 +92,14 @@ def test_identity_resample():
...
         rot_img = resample_img(Nifti1Image(data.astype('>f8'), affine),
                                target_affine=affine.tolist(),
                                interpolation=interpolation)
         np.testing.assert_almost_equal(data, rot_img.get_data())
 
     # Test with little endian data ('<f8')
-    for interpolation in ['nearest', 'continuous']:
+    for interpolation in ['nearest','linear','continuous']:
```
Comment (GaelVaroquaux) [inline nilearn/image/tests/test_resampling.py]: Same comment on style: space after comma.

**Comment by GaelVaroquaux** [comment]
> Thank's for the direct advice. I'm new to this :-)

Don't worry: we've all been new. Thanks a lot for pitching in, you're
providing value and we appreciate.

File: nilearn/image/tests/test_resampling.py (hunk 2 above)
Comment (jnecus) [reply nilearn/image/tests/test_resampling.py]: No worries, got to learn these things at some point :-)

**Comment by GaelVaroquaux** [comment]
By the way: our online PEP8 checker found that you had trailing whitespace (unused spaces at the end of a line): https://travis-ci.org/nilearn/nilearn/jobs/277296242

You should configure your editor to show them, so that you can remove them.

I'll remove them, don't worry about it.

**Comment by jnecus** [comment]
@GaelVaroquaux  Sorry for the unnecessary hassle and thanks for all the help with getting started, I appreciate it. Things should hopefully go more smoothly next time!

**Comment by GaelVaroquaux** [comment]
@jnecus : what's you're real-world name? To credit you.

**Comment by jnecus** [comment]
@GaelVaroquaux Joe Necus

**Comment by GaelVaroquaux** [comment]
OK, I added an entry in the whats_new file

**Comment by banilo** [comment]
+1

**Pull request merged**